modules/integration/src/test/scala/scala/cli/integration/RunTestDefinitions.scala

```diff
@@ -552,6 +552,59 @@ abstract class RunTestDefinitions(val scalaVersionOpt: Option[String])
     }
   }
 
+  test("project settings file") {
+    val inputs = TestInputs(
+      os.rel / "custom-dir" / "project.settings.scala" -> "",
+      os.rel / "project.settings.scala" -> s"//> using javaProp \"foo=bar\"".stripMargin,
+      os.rel / "foo.scala" ->
+        s"""object Foo {
+           |  def main(args: Array[String]): Unit =
+           |    println(sys.props("foo"))
+           |}
+           |""".stripMargin
+    )
+    inputs.fromRoot { root =>
+      // format: off
+      val out =
+        os.proc(
+          TestUtil.cli,
+          "custom-dir", 
+          "foo.scala", 
+          "project.settings.scala",
+          extraOptions)
+        .call(cwd = root).out.text().trim
+      // format: on
+      assert(out == "bar")
+      assert(os.exists(root / "custom-dir" / ".scala-build"))
+      assert(!os.exists(root / ".scala-build"))
+    }
+  }
+
+  test("setting root dir without project settings file") {
+    val inputs = TestInputs(
+      os.rel / "custom-dir" / "foo.scala" ->
+        s"""object Foo {
+           |  def main(args: Array[String]): Unit =
+           |    println("Foo")
+           |}
+           |""".stripMargin,
+      os.rel / "bar.scala" -> ""
+    )
+    inputs.fromRoot { root =>
+      os.proc(TestUtil.cli, "custom-dir", "bar.scala", extraOptions).call(cwd = root)
+      assert(os.exists(root / "custom-dir" / ".scala-build"))
+      assert(!os.exists(root / ".scala-build"))
+    }
+  }
```

Comment:
It seems that these tests can be moved to unit tests, for example into `InputsTests`.

One integration test is enough, e.g. the one test with gist you can keep in `RunTestsDefinitions`.